HttpServices/Services/Concrete/JsonRestClient.cs

```diff
@@ -68,11 +72,15 @@ private async Task<TResult> SubmitRequest<TResult>(string url, object payload, s
 
             if (payload != null)
             {
-                var jsonBody = JsonConvert.SerializeObject(payload);
-                httpRequestMessage.Content = new StringContent(jsonBody, Encoding.UTF8, "application/json");
+                var ms = new MemoryStream();
+                await JsonSerializer.SerializeAsync(ms, payload);
+                ms.Seek(0, SeekOrigin.Begin);
+                httpRequestMessage.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                using var requestContent = new StreamContent(ms);
```

Comment:
1. no using should be here - it will fail on sending as content is already disposed. If you implemented it based on post https://code-maze.com/using-streams-with-httpclient-to-improve-performance-and-memory-usage/, reproduce it carefully
2. json serialization / deserialization is typical CPU-bound operation, and it shouldn't be async on server (for CPU-bound operations is just switching thread on same CPU, so it's performance degradation for thread switching without benefit). So instead of SerializeAsync consider using of Serialize with writer:

```
var requestStream = new MemoryStream();
var writer = new Utf8JsonWriter(requestStream);

JsonSerializer.Serialize(writer, payload);
requestStream.Seek(0, SeekOrigin.Begin);
```


async serialization is aimed for GUI apps or batch serialization of many objects at once. See more at
- https://docs.microsoft.com/en-us/archive/msdn-magazine/2014/october/async-programming-introduction-to-async-await-on-asp-net#asynchronous-code-is-not-a-silver-bullet
- https://stackoverflow.com/a/30574578

3. There are 3 logical blocks in this code piece - serialization, setting headers, filling up request content, so help your fellow readers and separate them by empty lines :)

Reply:
Well, learnt something :)